PsfLauncher/main.cpp

```diff
@@ -268,43 +276,278 @@ void LaunchMonitorInBackground(std::filesystem::path packageRoot, const wchar_t
         }
         else
         {
-            //Log("error starting monitor using SellExecuteEx also. Error=0x%x\n", ::GetLastError());
+            auto err = ::GetLastError();
+            ErrorInformation error = { L"error starting monitor using ShellExecuteEx", err, executable };
+        }
+
+        DWORD exitCode = ERROR_SUCCESS;
+        if (!GetExitCodeProcess(shExInfo.hProcess, &exitCode))
+        {
+            return { L"Could not get the exit process for " + std::wstring(executable), GetLastError() };
+        }
+        else
+        {
+            if (exitCode != ERROR_SUCCESS)
+            {
+                return { L"The process " + std::wstring(executable) + L" exited with with an unsuccessful code.", exitCode };

```

Comment:
a non-zero process exit code does not necessarily imply failure (it's up to the process) - I'd just omit "unsuccessful"